lib/javascript/JavascriptParser.js

```diff
@@ -356,53 +361,149 @@ class JavascriptParser extends Parser {
 			.tap("JavascriptParser", _expr => {
 				const expr = /** @type {BinaryExpressionNode} */ (_expr);
 
-				const handleNumberOperation = fn => {
+				const handleConstOperation = fn => {
+					const left = this.evaluateExpression(expr.left);
+					if (!left || !left.isCompileTimeValue()) return;
+
+					const right = this.evaluateExpression(expr.right);
+					if (!right || !right.isCompileTimeValue()) return;
+
+					const result = fn(
+						left.asCompileTimeValue(),
+						right.asCompileTimeValue()
+					);
+					switch (typeof result) {
+						case "boolean":
+							return new BasicEvaluatedExpression()
+								.setBoolean(result)
+								.setRange(expr.range);
+						case "number":
+							return new BasicEvaluatedExpression()
+								.setNumber(result)
+								.setRange(expr.range);
+						case "bigint":
+							return new BasicEvaluatedExpression()
+								.setBigInt(result)
+								.setRange(expr.range);
+						case "string":
+							return new BasicEvaluatedExpression()
+								.setString(result)
+								.setRange(expr.range);
+					}
+				};
+
+				const isAlwaysDifferent = (a, b) =>
+					(a === true && b === false) || (a === false && b === true);
+
+				const handleTemplateStringCompare = (left, right, res, eql) => {
+					const getPrefix = parts => {
+						let value = "";
+						for (const p of parts) {
+							const v = p.asString();
+							if (v !== undefined) value += v;
+							else break;
+						}
+						return value;
+					};
+					const getSuffix = parts => {
+						let value = "";
+						for (let i = parts.length - 1; i >= 0; i--) {
+							const v = parts[i].asString();
+							if (v !== undefined) value = v + value;
+							else break;
+						}
+						return value;
+					};
+					const leftPrefix = getPrefix(left.parts);
+					const rightPrefix = getPrefix(right.parts);
+					const leftSuffix = getSuffix(left.parts);
+					const rightSuffix = getSuffix(right.parts);
+					const lenPrefix = Math.min(leftPrefix.length, rightPrefix.length);
+					const lenSuffix = Math.min(leftSuffix.length, rightSuffix.length);
+					if (
+						leftPrefix.slice(0, lenPrefix) !==
+							rightPrefix.slice(0, lenPrefix) ||
+						leftSuffix.slice(-lenSuffix) !== rightSuffix.slice(-lenSuffix)
+					) {
+						return res.setBoolean(!eql);
```

Comment:
not sure that this will not break the world.. since
```
`string1${sideEffect()}` === `1string${sideEffect()}` 
```
side effect should be executed